src/mapContainer.ts

```diff
@@ -645,13 +713,181 @@ export class MapContainer {
+                    color: this.settings.edgeColor,
+                    weight: 1,
+                });
+                edge.polyline = polyline;
+                polyline.addTo(this.display.map);
+                this.display.polylines.push(polyline);
+            }
+        }
+    }
+
+    private clearPolylines() {
+        // clear previous polylines from the map
+        for (const p of this.display.polylines) {
+            p.remove();
+        }
+        this.display.polylines.length = 0;
+    }
+
+    private generateEdgesFromFilesWithMarkers(
+        fileMarkerMap: Map<string, FileWithMarkers>
+    ): Map<string, Edge> {
+        let nodesSeen: Set<string> = new Set();
+        let edges: Map<string, Edge> = new Map();
+        for (let [k, fwm] of fileMarkerMap) {
+            this.generateEdgesFromFileWithMarkers(
+                fwm,
+                fileMarkerMap,
+                edges,
+                nodesSeen
+            );
+        }
+        return edges;
+    }
+
+    private generateEdgesFromFileWithMarkers(
+        source: FileWithMarkers,
+        fileMarkerMap: Map<string, FileWithMarkers>,
+        edges: Map<string, Edge>,
+        nodesSeen: Set<string>
+    ) {
+        const file = source.file;
+        const path = file.path;
+        if (nodesSeen.has(path)) {
+            return;
+        }
+        nodesSeen.add(path);
+        const fileCache = this.app.metadataCache.getFileCache(file);
+        for (let link of fileCache?.links || []) {
+            let destination = this.app.metadataCache.getFirstLinkpathDest(
+                link.link,
+                path
+            );
+            if (destination?.path && fileMarkerMap.has(destination.path)) {
+                // both the source file and the destination file have markers;
+                // let's connect them and create edges
+                let destinationFileWithMarkers = fileMarkerMap.get(
+                    destination.path
+                );
+                for (let sourceMarker of source.markers) {
+                    for (let destinationMarker of destinationFileWithMarkers.markers) {
+                        let edge: Edge = {
+                            v1: new leaflet.LatLng(
+                                sourceMarker.location.lat,
+                                sourceMarker.location.lng
+                            ),
+                            v2: new leaflet.LatLng(
+                                destinationMarker.location.lat,
+                                destinationMarker.location.lng
+                            ),
+                        };
+                        edges.set(
+                            `${edge.v1.toString()}<<->>${edge.v2.toString()}`,
+                            edge
+                        );
+                    }
+                }
+                // continue to traverse files and links recursively
+                this.generateEdgesFromFileWithMarkers(
```

Comment:
1. I think users will want this configurable, e.g. "depth of links to show", which can be between 1 (only direct links) and some arbitrary number.
2. Will this work correctly (i.e. not create an infinite recursion) in case of circles? What happens when links go A->B->C->A?

Reply:
@esm7, everything except this change has been pushed for your review. For this one, I need a little more detail on how exactly this would work. Could you provide an example graph (example: note_a -> note_b -> note_c; note_b -> note_e; etc.) and then explain how things would work if the new depth setting is set to 1 vs. set to 3? Thanks.

Reply:
The way I see it, you can just initially call this method with `settings.linkDepthForEdges` set to a function argument `linkDepth`, for every level of a recursion call it with `linkDepth - 1`, and when `linkDepth == 0` in the beginning of the function you return. So the recursion never goes into more than `settings.linkDepthForEdges` levels.

Reply:
ahh. I understand this change now. Cool setting. I'll make this change in the next few hours and push it @esm7.